src/Scorecard.ts

```diff
@@ -0,0 +1,56 @@
+import {CategoryScores} from "./domain";
+
+const ONES_THROUGH_SIXES_BONUS_SCORE = 63;
+
+export class Scorecard {
+  private chosenCategories: string[]
+  private scores: CategoryScores;
+  private yahtzeeBonus: boolean;
+
+  constructor() {
+    this.yahtzeeBonus = false;
+    this.chosenCategories = []
```

Comment:
Could we determine which categories are chosen from `this.scores`? There would need to be a way to distinguish between 0 as an unchosen category vs 0 as a chosen category but I'd argue if 0 is a valid score for a category that this distinction is better incorporated into `this.scores`. 

If `undefined` were used to indicate an unchosen category, I think `this.scores` is easier to read and doesn't rely on `this.chosenCategories` to interpret that 0.